// eslint/lib/rules/no-empty-static-block.js
/**
 * @fileoverview Rule to disallow empty static blocks.
 * @author Sosuke Suzuki
 */
"use strict";

//------------------------------------------------------------------------------
// Rule Definition
//------------------------------------------------------------------------------

/** @type {import('../shared/types').Rule} */
module.exports = {
	meta: {
		type: "suggestion",

		docs: {
			description: "Disallow empty static blocks",
			recommended: true,
			url: "https://eslint.org/docs/latest/rules/no-empty-static-block",
		},

		schema: [],

		messages: {
			unexpected: "Unexpected empty static block.",
		},
	},

	create(context) {
		const sourceCode = context.sourceCode;

		return {
			StaticBlock(node) {
				if (node.body.length === 0) {
					const closingBrace = sourceCode.getLastToken(node);

					if (
						sourceCode.getCommentsBefore(closingBrace).length === 0
					) {
						context.report({
							node,
							messageId: "unexpected",
						});
					}
				}
			},
		};
	},
};

// These are personal recreations of existing projects, developed by Ashraf Morningstar for learning and skill development. Original project concepts remain the intellectual property of their respective creators.
https://github.com/AshrafMorningstar
